src/bin/pgcopydb/ld_stream.c

```diff
@@ -35,6 +35,150 @@
 static bool updateStreamCounters(StreamContext *context,
 								 LogicalMessageMetadata *metadata);
 
+/*
+ * Determines whether a message needs to be filtered out based on the
+ * provided filters. The caller function should have already parsed and
+ * validated the filters.
+ */
+bool
+ShouldFilterOutMessage(SourceFilters *filters, char *nspname, char *relname)
+{
+	/*
+	 * Validate nspname is not NULL or empty
+	 */
+	if (nspname == NULL || strlen(nspname) == 0)
```

Comment:
Use IS_EMPTY_STRING_BUFFER to check whether the given C string is empty.
```suggestion
	if (nspname == NULL || IS_EMPTY_STRING_BUFFER(nspname))
```